extensions/emmet/src/abbreviationActions.ts

```diff
@@ -139,7 +139,12 @@ export function expandEmmetAbbreviation(args): Thenable<boolean> {
 		return [new vscode.Range(abbreviationRange.start.line, abbreviationRange.start.character, abbreviationRange.end.line, abbreviationRange.end.character), abbreviation, filter];
 	};
 
-	editor.selections.forEach(selection => {
+	let selectionsInReverseOrder = editor.selections.slice(0);
+	selectionsInReverseOrder.sort((a, b) => {
+		return a.anchor.compareTo(b.anchor) * -1;
```

Comment:
What we should compare here is the ending position of the abbreviation

`anchor` comes after `active` when `selection.isReversed` is true.

So instead of `a.anchor.compareTo(b.anchor)`, this should probably be 
`(a.isReversed ? a.anchor: a:active).compareTo(b.isReversed ? b.anchor: b:active)` ?

Reply:
@jmdowns2 ?